src/OSPSuite.Presentation/Presenters/Charts/ChartEditorPresenter.cs

```diff
@@ -240,15 +247,80 @@ public class ChartEditorPresenter : AbstractCommandCollectorPresenter<IChartEdit
          _axisSettingsPresenter.AxisAdded += (o, e) => onAxisAdded();
          _axisSettingsPresenter.AxisPropertyChanged += (o, e) => updateChart();
 
-         AddSubPresenters(axisSettingsPresenter, chartSettingsPresenter, chartExportSettingsPresenter, curveSettingsPresenter, dataBrowserPresenter);
+         _curveColorGroupingPresenter.ApplySelectedColorGrouping += (o, e) => onApplyColorGrouping(e.SelectedMetaData);
+
+         AddSubPresenters(axisSettingsPresenter, chartSettingsPresenter, chartExportSettingsPresenter, curveSettingsPresenter, dataBrowserPresenter, curveColorGroupingPresenter);
 
          _view.SetAxisSettingsView(axisSettingsPresenter.View);
          _view.SetChartSettingsView(chartSettingsPresenter.View);
          _view.SetChartExportSettingsView(chartExportSettingsPresenter.View);
          _view.SetCurveSettingsView(curveSettingsPresenter.View);
+         _view.SetCurveColorGroupingView(curveColorGroupingPresenter.View);
          _view.SetDataBrowserView(dataBrowserPresenter.View);
       }
 
+      //gets all the common metaData of the observed data that correspond to active curves 
+      private IReadOnlyList<string> getCommonMetaDataOfCurves(IReadOnlyList<DataRepository> allObservedData)
+      {
+         var activeDataColumns = _dataBrowserPresenter.GetAllUsedDataColumns();
+         _activeObservedData = allObservedData.Where(x => activeDataColumns.Any(column => column.Repository.Name == x.Name)).ToList();
+
+         if (_activeObservedData == null || !_activeObservedData.Any())
+            return new List<string>();
+
+         var firstCurveMetaData = _activeObservedData.First().ExtendedProperties.Keys.ToList();
+         var commonMetaData = firstCurveMetaData.Where(x => _activeObservedData.All(observedData => observedData.ExtendedProperties.Keys.Contains(x))).ToList();
+
+         return commonMetaData;
+      }
+
+      private void onApplyColorGrouping(IReadOnlyList<string> eSelectedMetaData)
+      {
+         if (_activeObservedData == null)
```

Comment:
If you defined this list as a readonly list, yopu never have to check for null. AT most you clear, and you AddRange when you update it. Last, call it _activeOBservedDataList because it looks like it's only one from the naming

Reply:
True, I changed this